--- a/libgitg/gitg-diff-view-file-info.vala
+++ b/libgitg/gitg-diff-view-file-info.vala
@@ -1,4 +1,4 @@
-class Gitg.DiffViewFileInfo : Object
+public class Gitg.DiffViewFileInfo : Object
 {
 	public Ggit.DiffDelta delta { get; construct set; }
 	public bool from_workdir { get; construct set; }
@@ -93,7 +93,7 @@
 		}
 	}
 
-	private async void guess_content_type(InputStream stream, string basename, Cancellable? cancellable)
+	private async void guess_content_type(InputStream stream, string? basename, Cancellable? cancellable)
 	{
 		var buffer = new uint8[4096];
 		size_t bytes_read = 0;
