--- a/libgitg/gitg-diff-view.vala
+++ b/libgitg/gitg-diff-view.vala
@@ -72,11 +72,6 @@
 		}
 	}
 
-	public bool has_selection
-	{
-		get; private set;
-	}
-
 	public Ggit.Diff? diff
 	{
 		get { return d_diff; }
@@ -639,25 +634,6 @@
 		SignalHandler.unblock(d_commit_details, d_expanded_notify);
 	}
 
-	private void on_selection_changed()
-	{
-		bool something_selected = false;
-
-		foreach (var file in d_grid_files.get_children())
-		{
-			if (((Gitg.DiffViewFile) file).has_selection())
-			{
-				something_selected = true;
-				break;
-			}
-		}
-
-		if (has_selection != something_selected)
-		{
-			has_selection = something_selected;
-		}
-	}
-
 	private string? primary_path(Ggit.DiffDelta delta)
 	{
 		var path = delta.get_old_file().get_path();
@@ -850,10 +826,8 @@
 								bind_property("wrap-lines", renderer_text, "wrap-lines", BindingFlags.DEFAULT | BindingFlags.SYNC_CREATE);
 								bind_property("tab-width", renderer_text, "tab-width", BindingFlags.DEFAULT | BindingFlags.SYNC_CREATE);
 								renderer_text.maxlines = maxlines;
-								renderer_text.notify["has-selection"].connect(on_selection_changed);
 							}
 						}
-						on_selection_changed();
 					}
 					if (current_is_binary)
 					{
@@ -1050,26 +1024,6 @@
 		auto_change_expanded(true);
 	}
 
-	public PatchSet[] get_selection()
-	{
-		var ret = new PatchSet[0];
-
-		foreach (var file in d_grid_files.get_children())
-		{
-			ret += ((Gitg.DiffViewFile)file).get_selection();
-		}
-
-		return ret;
-	}
-
-	public void clear_selection()
-	{
-		foreach (var file in d_grid_files.get_children())
-		{
-			((Gitg.DiffViewFile)file).clear_selection();
-		}
-	}
-
 	private void update_hide_show_options(Gdk.Window window, int ex, int ey)
 	{
 		void *data;
