--- a/libgitg/gitg-init.vala
+++ b/libgitg/gitg-init.vala
@@ -66,19 +66,6 @@
 	factory.register(typeof(Ggit.Commit),
 	                 typeof(Gitg.Commit));
 
-	factory.register(typeof(Ggit.Remote),
-	                 typeof(Gitg.Remote));
-
-	// Add our own css provider
-	Gtk.CssProvider? provider = Gitg.Resource.load_css("libgitg-style.css");
-
-	if (provider != null)
-	{
-		Gtk.StyleContext.add_provider_for_screen(Gdk.Screen.get_default(),
-		                                         provider,
-		                                         600);
-	}
-
 }
 
 }
